src/tests/end-to-end/tests/target-page/issue-dialog.test.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { Browser } from '../../common/browser';
+import { launchBrowser } from '../../common/browser-factory';
+import { PopupPage } from '../../common/page-controllers/popup-page';
+import { TargetPage } from '../../common/page-controllers/target-page';
+import { scanForAccessibilityIssues } from '../../common/scan-for-accessibility-issues';
+
+describe('Target Page issue dialog', () => {
+    let browser: Browser;
+    let targetPage: TargetPage;
+    let popupPage: PopupPage;
+
+    beforeAll(async () => {
+        browser = await launchBrowser({ suppressFirstTimeDialog: true });
+        targetPage = await browser.newTargetPage();
+        popupPage = await browser.newPopupPage(targetPage);
+    });
+
+    afterAll(async () => {
+        if (browser) {
+            await browser.close();
+            browser = undefined;
+        }
+    });
+
+    it('should pass accessibility validation', async () => {
+        await popupPage.gotoAdhocPanel();
+        await popupPage.enableToggleByAriaLabel('Automated checks');
+
+        await targetPage.bringToFront();
+
+        const shadowRoot = await targetPage.getShadowRoot();
+        const issueHighlightLabel = await shadowRoot.$('.failure-label');
```

Comment:
I don't think this is waiting for the `.failure-label` to appear; I think it's assuming it's already appeared. Probably safer to use a wait-based operation.

`ElementHandle` doesn't expose a `waitForSelector` directly like `Page` does, but you can emulate one with `waitForFunction`; probably would be a good thing to add to our `Page` wrapper, something like:

```ts
async function waitForDescendentSelector(parentElement: ElementHandle<Element>, descendentSelector: string, options?: WaitOptions) {
    options = {
        timeout: DEFAULT_TIMEOUT_CONSTANT,
        ...options
    };
    return await this.underlyingPage.waitForFunction((parent, selector) => parent.querySelector(selector), options, parentElement, descendentSelector);
}
```

Reply:
I was trying to get the element that I would click, but I agree that it is safer to wait for it to appear before doing so.